woocommerce/woocommerce
# Shipping Method Settings: Update modal contents

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/40189

Update modal contents for Shipping Zone Methods and Shipping Classes according to designs found in paJDVv-2Ew-p2.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Head to `/wp-admin/admin.php?page=wc-settings&tab=shipping` on a Jurassic Ninja site with this branch
2. Add a Shipping Zone with Name and Regions added.
3. Click "Add shipping method" and go through the flow for Flat Rate, Free Shipping and Local Pickup the modals should look like the designs.
5. Be sure to select a few options for `FREE SHIPPING REQUIRES` to see the rest of the form change dynamically.
6. Now add a Shipping Class from `/wp-admin/admin.php?page=wc-settings&tab=shipping&section=classes`. That modal should also reflect the new designs.
7. Now that you've added a Shipping Zone Class, return to the Shipping Zone Methods screen. Add or edit a Flat Rate method, you should now see the configuration modal include fields pertinent to your Shipping Class.
8. Now add a new method from an extension. [Flat Rate Boxes](https://github.com/woocommerce/woocommerce-shipping-flat-rate-boxes/releases), for example. Add a new method Flat Rate Boxes. Note, it doesn't have a configuration screen, so the second modal is omitted.

### Known issues

The following are known and will be addressed in a follow up.

Between Adding and configuring a Shipping Method, the button's spinner needs help
<img width="100" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1922453/cfc83529-903c-4f7c-b4d9-86b87b209683">

After a Shipping Class is added, configuring a Flat Rate method includes new fields. Since we programatically append the help text underneath the input, these look off since they are so long. @FlorianBru should we go back to the help tooltip for these or simply change the text?

<img width="565" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1922453/1c18535f-cb8d-4a5b-8ef5-010e77aa8872">





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Update Shipping Class modals to new style

#### Comment 

</details>
